test/Microsoft.Azure.Devices.Edge.Test.Common/TomlDocument.cs

```diff
@@ -24,67 +22,64 @@ public void ReplaceOrAdd<T>(string dottedKey, T value)
             {
                 string tableKey = segments[i];
 
-                try
+                if (!table.ContainsKey(tableKey))
                 {
-                    table = (TomlTable)table[tableKey];
-                }
-                catch (KeyNotFoundException)
-                {
-                    // Nett does not provide a function to easily add table subkeys.
-                    // A hack workaround is to serialize the table, append the new table as a string, then deserialize.
-                    // This only needs to be done once to create the new subtable.
-                    // After that, Nett functions can be used to modify the subtable.
-                    string tableName = string.Join(".", segments.Take(segments.Length - 1));
-                    this.AddTable(tableName, segments[segments.Length - 1], value);
-                    return;
+                    if (add)
+                    {
+                        table.Add(tableKey, table.CreateEmptyAttachedTable());
+                    }
+                    else
+                    {
+                        return (table, string.Empty);
+                    }
                 }
+
+                table = (TomlTable)table[tableKey];
             }
 
-            string key = segments[segments.Length - 1];
+            return (table, segments[segments.Length - 1]);
+        }
+
+        public void ReplaceOrAdd(string dottedKey, bool value)
+        {
+            var (table, key) = this.TraverseKey(dottedKey, add: true);
```

Comment:
Just curious, what does the linter say about `add: true`?

Reply:
Since `add` is an optional parameter, the linter might have suggested this, but I don't remember for sure. It definitely didn't complain. I did this a while ago, but I think I added the explicit parameter name here because if you just see a call to
```csharp
this.TraverseKey(dottedKey, true);
```
...it's not immediately obvious what the "true" argument is for.